internal/providers/azure/azure.go

```diff
@@ -70,7 +70,19 @@ func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config,
 
 	logger := f.Logger
 	logger.Debug("waiting for config DVD...")
-	waitForCdrom(logger, devicePath)
+
+	// Look for the gen2 settings if avaialble
+	// else fall back to gen1 settings
+	device, err := execUtil.GetUdfBlockDevices()
+	if err != nil {
```

Comment:
This is a little tricky here. I would like to revisit this logic to understand if there's a better way to handle both gen1/gen2 settings using the `device` slice.